libs/wingsdk/src/target-sim/website.inflight.ts

```diff
@@ -0,0 +1,78 @@
+import { Server } from "http";
+import { AddressInfo } from "net";
+import express from "express";
+import { ApiAttributes, WebsiteSchema, WEBSITE_TYPE } from "./schema-resources";
+import { TraceType, IWebsiteClient } from "../cloud";
+import { Json } from "../std";
+import {
+  ISimulatorContext,
+  ISimulatorResourceInstance,
+} from "../testing/simulator";
+
+const LOCALHOST_ADDRESS = "127.0.0.1";
+
+export class Website implements IWebsiteClient, ISimulatorResourceInstance {
+  private readonly context: ISimulatorContext;
+  private readonly app: express.Application;
+  private server: Server | undefined;
+  private url: string | undefined;
+
+  constructor(props: WebsiteSchema["props"], context: ISimulatorContext) {
+    this.context = context;
+
+    // Set up an express server that handles the routes.
+    this.app = express();
+
+    // Use static directory
+    this.app.use(express.static(props.path));
+    this.initiateJsonRoutes(props.jsonRoutes);
+  }
+
+  private initiateJsonRoutes(routes: Record<string, Json>) {
+    for (const path in routes) {
+      this.app.get(`/${path}`, (_, res) => {
+        res.json(routes[path]);
+      });
+    }
+  }
+
+  public async init(): Promise<ApiAttributes> {
+    // `server.address()` returns `null` until the server is listening
+    // on a port. We use a promise to wait for the server to start
+    // listening before returning the URL.
+    const addrInfo: AddressInfo = await new Promise((resolve, reject) => {
+      this.server = this.app.listen(0, LOCALHOST_ADDRESS, () => {
```

Comment:
(thinking outloud) I think staying away from a fixed port is probably better for now, otherwise we might need to consider how to increment ports based on number of sites, and if port is in use do we fail or keep incrementing ports to find a valid `next` port. Also we may end up in some race conditions if we start to get opinionated on ports. 

Thought I do agree the port number changing on sim restarts could be frustrating.

Reply:
I think the user should choose the port (similar to when they make one from scratch) - do we know what target we are running? 
(so we can do something like `let w = new website({path: "...", domain: target == "sim" ? "localhost:8080" :  "real-domain.com" })`)